Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+# Package structure {#package-structure}
+
+```{r, include = FALSE}
+source("common.R")
+```
+
+This chapter will start you on the road to package development by converting the implicit knowledge you've gained from *using* R packages into the explicit knowledge needed to *create and modify* them. You'll learn about the various states a package can be in and the difference between a package and library (and why you should care). Finally, you'll get a blueprint for creating your own packages *de novo*.
+
+## What is a package? {#package}
+
+When you create or modify a package, you work on its "source code" or "source files". You interact with the in-development package in its *source form*. This is not the package form you are most familiar with from day-to-day usage. Package development workflows make much more sense if you understand the five states an R package can be in:
+
+  * source
+  * bundled
+  * binary
+  * installed
+  * in-memory
+  
+You already know some of the functions that put packages into these states. For example, `install.packages()` and `devtools::install_github()` move a package into the installed state, from one of the forms earlier in the list. The `library()` function loads an installed package into memory, making it available for immediate and direct use.
+
+### Source package
+
+A __source__ package is just a directory of files, with a very specific structure. It includes particular components, such as a `DESCRIPTION` file, an `R/` directory containing `.R` files, and so on. Most of the remaining chapters in this book are dedicated to detailing these components.
+
+If you are new to package development, you may have never seen a package in source form! You might not even have any source packages on your computer. The easiest way to see a package in source form right away is to browse around its code on the web.
+
+Many R packages are developed in the open on GitHub (or GitLab or similar). The best case scenario is that you visit the package's CRAN landing page, e.g.:
+
+  - forcats: <https://cran.r-project.org/package=forcats>
+  - stringr: <https://cran.r-project.org/package=stringr>
```

Comment:
In hindsight, I wish I'd chosen one simple package (forcats) and one with compiled code (readxl?). Might go back and do this.

Reply:
Yeah, I think that's worth it.

Reply:
Done.